avm/res/network/load-balancer/inbound-nat-rule/main.bicep

```diff
@@ -51,15 +51,15 @@ param idleTimeoutInMinutes int = 4
 ])
 param protocol string = 'Tcp'
 
-resource loadBalancer 'Microsoft.Network/loadBalancers@2023-04-01' existing = {
+resource loadBalancer 'Microsoft.Network/loadBalancers@2023-11-01' existing = {
   name: loadBalancerName
 }
 
-resource inboundNatRule 'Microsoft.Network/loadBalancers/inboundNatRules@2023-04-01' = {
+resource inboundNatRule 'Microsoft.Network/loadBalancers/inboundNatRules@2023-11-01' = {
   name: name
   properties: {
-    frontendPort: frontendPort
-    backendPort: backendPort
+    frontendPort: frontendPort != -1 ? frontendPort : null
+    backendPort: backendPort != -1 ? backendPort : frontendPortRangeStart != -1 ? frontendPortRangeStart : null
```

Comment:
I suggest to update both parameters to nullable values with nullable checks (if any).
In case of the `backendPort` it might be enough to write `backendPort ?? frontendPortRangeStart` (i.e., use the first non-null value)